### ISSUE TITLE
Worker fails to poll database for messages after loosing the DB connection for certain period of time which exceeds pollInterval

### ISSUE THREAD

**Issue by Siva-R** [graphile/worker#185]
### Summary

After the system which is running worker disconnects from database and if it exceeds the pollInterval by considerable amount of time, it fails to poll the database. We have to restart the worker in order to catchup the new messages.

### Steps to reproduce

1) Start the graphile worker with a pollInterval of 1000 ms and ensure that it is connected to the database.
2) After a while turn off the wifi to disconnect from DB and make sure that worker is in this state for 10 minutes
3) Turn on the wifi, you will see that worker has stopped polling the database and it will not receive the new messages.

### Expected results
Worker should be able to receive the new messages after it reconnects to the database.

### Actual results
Worker fails to receive the messages.

**Comment by Siva-R** [comment]
Hi Benjie, did you get a chance to look at this issue?

**Comment by benjie** [comment]
Please can you include the console messages the worker is outputting in this situation? You should be seeing messages such as `Failed to acquire job:...` A full log output would be appreciated.

**Comment by benjie** [comment]
This may be working as designed; there's a maximum contiguous error count which if exceeded the worker shuts down:

```ts
  /**
   * How many errors in a row can we get fetching a job before we raise a higher
   * exception?
   */
  maxContiguousErrors: number;
```

The logs should let you know if this is happening.

**Comment by Siva-R** [comment]
Thanks. Is there a way to configure maxContiguousErrors in options passed to run method ?

**Comment by benjie** [comment]
Yes, you can pass it as `maxContiguousErrors` anywhere cosmicconfig supports.

[semi-automated message] Thanks for your question; hopefully we're well on the way to helping you solve your issue. This doesn't currently seem to be a bug in the library so I'm going to close the issue, but please feel free to keep requesting help below and if it does turn out to be a bug we can definitely re-open it 👍

You can also ask for help in the #help-and-support channel in [our Discord chat](http://discord.gg/graphile).
